polaris/hub/polarisfs.py

```diff
@@ -212,3 +232,15 @@ def pipe_file(
             timeout=timeout,
         )
         response.raise_for_status()
+
+        new_listing = [{"name": path, "type": "file", "size": len(content)}]
+        try:
+            current_listings = self._ls_from_cache(self._parent(path))
+            new_listing.extend(current_listings)
+        except FileNotFoundError:
+            # self._ls_from_cache() raises FileNotFoundError when no listings
+            # are found within a path. We can then assume it's a new directory
+            # and new_listing can be used as is.
+            pass
```

Comment:
`extend` is not in-place, so this actually doesn't have an effect.
```suggestion
        new_listing = [{"name": path, "type": "file", "size": len(content)}]
        try:
            current_listings = self._ls_from_cache(self._parent(path))
            new_listing = current_listings.extend(new_listing)
        except FileNotFoundError:
            # self._ls_from_cache() raises FileNotFoundError when no listings
            # are found within a path. We can then assume it's a new directory
            # and new_listing can be used as is.
            pass
```

Reply:
I believe `extend` does modify in place. Doing this way returns `None` for `new_listing`.

Reply:
You're absolute right. My bad!